protobuf/test/BUILD

```diff
@@ -0,0 +1,19 @@
+cc_test(
+    name = 'test_protos_cpp',
+    srcs = ['test_protos.cpp'],
+    deps = ['//protobuf/protos:protos_cc',
+            '@googletest//:gtest_main'],
+)
+
+py_test(
+    name = 'test_protos_py',
+    srcs = ['test_protos.py',
+            'test_protos_v2.py',
+            'py_tests_main.py',
+            ],
+    main = 'py_tests_main.py',
+    deps = [
+            '//protobuf/protos:protos_pkg',
```

Comment:
this depends on the package created by the new rule

Reply:
Oh pretty clever that you can still use the normal `py_test` rule for this! But... What about dependencies? Also, at this point the package isn't really "installed" in the pip sense of the word; this is also probably why you still need to modify the working directory in `py_tests_main.py`.

Lastly, I think this still wouldn't work when this package has dependencies that are similarly built with this rule?

Reply:
should work as well because the paths/imports for the transitive dependencies should be added in the runpath (and if they're not then I made a mistake in the rule :) ).
For installing with pip or similar, the problem is running in a sandbox without installing in the host python.
If there is another way, happy to explore it

Reply:
for the working directory, I needed to modify it because of the automatic discovery performed by `nose`. Since by default it runs in the root, it would find _every_ test in the repository and external packages O__O

Reply:
I have to admit I've been lazy on that part; instead of learning the API of nose I just changed the entry point :)

Reply:
I thought pytest was all the rage these days ;) Works with Bazel too, already tried it.

Regarding your comment on the transitive dependencies, that would be great! Have you already tested that? I commented on it because earlier we concluded that the preferred approach would be to install first (potentially in a venv), followed by testing, to ensure the installation actually works as intended. But I think it's a minor point. If this works I'll gladly take it!

Reply:
added an example of a module that uses the protos and how to use it in unit tests.

This rule will not create a package that you can install with `pip install`, just for clarity :), but  combining "standard" python modules/libraries with the proto package will be supported by bazel, as shown in the example called `test_consumer.py`